src/uu/tail/src/tail.rs

```diff
@@ -270,7 +270,11 @@ impl Error for ParseSizeErr {
 
 impl fmt::Display for ParseSizeErr {
     fn fmt(&self, f: &mut fmt::Formatter) -> Result<(), fmt::Error> {
-        write!(f, "{}", self.to_string())
+        let s = match self {
```

Comment:
Actually this should probably just format as Debug instead of effectively reimplementing it.  The main takeaway is not to use self.to_string() from within fmt!